elasticgraph-graphql/lib/elastic_graph/graphql/filtering/filter_interpreter.rb

```diff
@@ -339,7 +339,8 @@ def process_list_count_expression(bool_node, expression, field_path)
         end
 
         def build_bool_hash(&block)
-          bool_node = Hash.new { |h, k| h[k] = [] }.tap(&block)
+          bool_node = Hash.new { |h, k| h[k] = [] } #: stringOrSymbolHash
```

Comment:
The most impactful change in Steep 1.10 would be this. It inferrers correct types for some method calls with blocks, which results in reporting more type errors with this pattern. Now the type of `h` and `k` are not `untyped` but `Hash[K, V]` and `V`, and type errors are detected around them.

We need some type annotations here. (And most of `Hash.new` calls through the codebase.)